app/src/components/sponsoredTokenCard.tsx

```diff
@@ -13,10 +13,17 @@ export type SponsoredTokenCardProps = {
 export const SponsoredTokenCard = (props: SponsoredTokenCardProps): React.ReactElement => {
   const { notdClient } = useGlobals();
   const [asset, setAsset] = React.useState<RegistryToken | null>(null);
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<boolean>(false);
 
   const updateAsset = React.useCallback(async (): Promise<void> => {
+    setIsLoading(true);
     notdClient.retrieveRegistryToken(props.token.registryAddress, props.token.tokenId).then((registryToken: RegistryToken): void => {
       setAsset(registryToken);
+      setIsLoading(false);
+    }).catch(() => {
```

Comment:
catch the error and use that with `setError` rather than just setting true/false